spec/lib/vmdb/settings_spec.rb

```diff
@@ -566,6 +566,47 @@
       settings = Vmdb::Settings.for_resource(MiqServer.new(:zone => server.zone))
       expect(settings.api.token_ttl).to eq "4.hour"
     end
+
+    context "on global region" do
+      let(:key)                 { "yuri" }
+      let(:value_region_remote) { "region_remote" }
+      let(:value_server_remote) { "server_remote" }
+
+      before do
+        Zone.seed
+        @region_global = MiqRegion.first
+        @region_remote = FactoryBot.create(:miq_region, :id => ApplicationRecord.id_in_region(1, @region_global.region + 1))
+        @zone_remote   = FactoryBot.create(:zone, :id => ApplicationRecord.id_in_region(2, @region_remote.region))
+        @server_remote = FactoryBot.create(:miq_server, :zone => @zone_remote, :id => ApplicationRecord.id_in_region(1, @region_remote.region))
+        SettingsChange.create!(:id            => ApplicationRecord.id_in_region(1, @region_global.region),
+                               :resource_id   => @region_global.id,
+                               :resource_type => "MiqRegion",
+                               :key           => "/#{key}",
+                               :value         => "value from global region")
+      end
+
+      after do
+        SettingsChange.delete_all
+      end
+
+      it "applies settings from remote sever if there are specified" do
+        SettingsChange.create!(:id            => ApplicationRecord.id_in_region(1, @region_remote.region),
+                               :resource_id   => @server_remote.id,
+                               :resource_type => "MiqServer",
+                               :key           => "/#{key}",
+                               :value         => value_server_remote)
+        expect(Vmdb::Settings.for_resource(@server_remote)[key]).to eq(value_server_remote)
+      end
+
+      it "applies settings from remote region if settings on remote server not specified" do
```

Comment:
Can you add a test for the zone as well...might as well cover all 3 levels.

Reply:
👍   done